beehive-lab/TornadoVM
# [api][design] Make TornadoVM API interfaces sealed

#### Description

Expanding on the suggestion of @SirYwell (a TornadoVM committer), this PR adds sealed classes for the main collections and types interfaces of TornadoVM.
This has a few benefits:
- Since main classes./interfaces are sealed, developers cannot create custom data types outside of the control of the TornadoVM JIT compiler and the runtime. 
- It simplifies internal checks (runtime) 

#### Backend/s tested

Mark the backends affected by this PR.


#### OS tested

Mark the OS where this PR is tested.


#### Did you check on FPGAs?

If it is applicable, check your changes on FPGAs.


#### How to test the new patch?

This PR updates the design of the APIs. It does not change behaviour. To check everything runs as expected:

```bash
// OpenCL Backend
$ make 
$ make tests

// SPIR-V Backend
$ make BACKEND=spirv
$ make tests

// PTX Backend
$ make BACKEND=ptx
$ make tests
```
